geozero/src/mvt/mvt_reader.rs

```diff
@@ -35,38 +38,35 @@ fn process_properties(
     processor.properties_begin()?;
     for (i, pair) in feature.tags.chunks(2).enumerate() {
         let [key_idx, value_idx] = pair else {
-            return Err(GeozeroError::Feature(format!(
-                "invalid feature.tags length: {:?}",
-                feature.tags.len()
-            )));
+            return Err(MvtError::InvalidFeatureTagsLength(
+                feature.tags.len()))?
         };
         let key = layer
             .keys
             .get(*key_idx as usize)
-            .ok_or_else(|| GeozeroError::Feature(format!("invalid key index {key_idx}")))?;
+            .ok_or(MvtError::InvalidKeyIndex(*key_idx))?
+            .clone();
```

Comment:
why introduce a `clone()` here? Seems it caused all the ref requirements below, introducing a lot more unneeded changes?

Reply:
TBH I can't remember why I thought that was a good, I'd gotten a bit confused with the ownership of the strings passed into the errors :confused:

Reply:
np, glad we caught it - it would have been a significant perf cost